src/main/java/mate/academy/Main.java

```
@@ -1,16 +1,44 @@
 package mate.academy;
 
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import mate.academy.lib.Injector;
+import mate.academy.model.CinemaHall;
 import mate.academy.model.Movie;
+import mate.academy.model.MovieSession;
+import mate.academy.service.CinemaHallService;
 import mate.academy.service.MovieService;
+import mate.academy.service.MovieSessionService;
 
 public class Main {
+    private static final Injector injector = Injector.getInstance("mate.academy");
+
     public static void main(String[] args) {
-        MovieService movieService = null;
+        MovieService movieService = (MovieService) injector.getInstance(MovieService.class);
 
         Movie fastAndFurious = new Movie("Fast and Furious");
         fastAndFurious.setDescription("An action film about street racing, heists, and spies.");
         movieService.add(fastAndFurious);
         System.out.println(movieService.get(fastAndFurious.getId()));
         movieService.getAll().forEach(System.out::println);
+
+        CinemaHallService cinemaHallService = (CinemaHallService) injector
+                .getInstance(CinemaHallService.class);
+
+        CinemaHall cinemaHall = new CinemaHall();
+        cinemaHall.setDescription("Hall 1");
+        cinemaHallService.add(cinemaHall);
+        System.out.println(cinemaHallService.get(cinemaHall.getId()));
+        cinemaHallService.getAll().forEach(System.out::println);
+
+        MovieSessionService movieSessionService = (MovieSessionService) injector
+                .getInstance(MovieSessionService.class);
+
+        MovieSession movieSession = new MovieSession(fastAndFurious, cinemaHall,
+                LocalDateTime.of(2025, 7, 25, 20, 15));
+        movieSessionService.add(movieSession);
+        System.out.println(movieSessionService.get(movieSession.getId()));
+        movieSessionService.findAvailableSession(1L, LocalDate.of(2025, 7, 25))
```

Potential issue: The method `findAvailableSession` is called with a hardcoded value `1L` for the movie ID. According to best practices and likely checklist requirements, you should use `fastAndFurious.getId()` instead of a hardcoded value to ensure the correct movie session is found and to avoid errors if the ID is not 1.